base/compiler/ssair/queries.jl

```diff
@@ -49,6 +49,14 @@ function stmt_effect_free(@nospecialize(stmt), @nospecialize(rt), src, sptypes::
                 eT ⊑ fT || return false
             end
             return true
+        elseif head === :new_opaque_closure
+            a = ea[1]
+            typ = argextype(a, src, sptypes)
+            # `Expr(:new)` of unknown type could raise arbitrary TypeError.
```

Comment:
```suggestion
            # TODO: model full error checks of jl_new_opaque_closure_jlcall / jl_new_opaque_closure
```

Aside: I seem to notice :splatnew isn't in this list, but I suppose we convert that to :new if we could determine it was error-free?

Reply:
Yes, I noticed that too. I have some follow on work that goes slightly in this direction.